cluster-autoscaler/metrics/liveness_test.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func getTestResponse(timeout time.Duration, checkMonitoring bool) *httptest.ResponseRecorder {
+	req := httptest.NewRequest("GET", "/health-check", nil)
+	w := httptest.NewRecorder()
+	healthCheck := NewHealthCheck(timeout)
+	if checkMonitoring {
+		healthCheck.StartMonitoring()
+	}
+	healthCheck.ServeHTTP(w, req)
+	return w
+}
+
+func TestOkServeHTTP(t *testing.T) {
+	w := getTestResponse(time.Second, true)
+	assert.Equal(t, 200, w.Code)
+}
+
+func TestFailServeHTTP(t *testing.T) {
+	w := getTestResponse(time.Nanosecond, true)
+	assert.Equal(t, 500, w.Code)
```

Comment:
Is this just setting a nanosecond timeout and relying on execution taking more that 1ns? Relying on timing of anything in unittest feels very wrong to me, we should simulate whatever timing we want to test by explicitly setting timestamps.